test/helpers.js

```diff
@@ -0,0 +1,52 @@
+import { writeFileSync, mkdtempSync } from 'fs';
+import path from 'path';
+
+const TEMP_DIR_PREFIX = '.mockedTests-';
+const writeJSONFile = (filename, data) =>
+  writeFileSync(filename, JSON.stringify(data));
+
+/**
+ * There's some magic in order to create a relative path to the temporal directory. but mostly it does 3 things.
+ * 1) gets the dirname of the given path.
+ * 2) finds the relative from the temp file to the dirname found in step 1)
+ * 3) joins relativePath to the remaining fullPath (minus the dirnamePath)
+ */
+
+const getRelocatedPath = (dir, fullPath) => {
+  const dirnamePath = path.dirname(fullPath);
+  const relativePath = path.relative(dir, dirnamePath);
+  return path.join(path.join(relativePath, fullPath.split(dirnamePath)[1]));
+};
+
+export const temporaryConfigDirectory = ({
+  rules = null,
+  customRulePaths = [],
+  schemaPaths = [],
+}) => {
+  const configDirectory = mkdtempSync(TEMP_DIR_PREFIX);
```

Comment:
I naively assumed `mkdtemp` would automatically write to the OS's tmpdir, but looks like I'm incorrect given your change to `.gitignore`.

I did find something interesting in the docs.

I believe we want something like this:

```suggestion
  const configDirectory = mkdtempSync(path.join(os.tmpdir(), TEMP_DIR_PREFIX));
```

That should allow us to remove the `.gitignore` and not have the test suite pollute our local env. 😄